packages/react-tabster/src/hooks/useFocusIndicatorStyle.ts

```diff
@@ -6,11 +6,21 @@ const defaultStyleRule = (theme: Theme) => ({
   outline: `solid 1px ${theme.alias.color.neutral.neutralForeground1}`,
 });
 
+export interface CreateFocusIndicatorStyleRuleOptions {
+  selector?: 'focus' | 'focus-within';
+}
+
+const defaultOptions: CreateFocusIndicatorStyleRuleOptions = {
+  selector: 'focus',
+};
+
 export const createFocusIndicatorStyleRule = (
   rule: MakeStylesStyleRule<Theme> = defaultStyleRule,
+  options: CreateFocusIndicatorStyleRuleOptions = defaultOptions,
```

Comment:
while this might be appealing idea, we introduce more complicated code

I'd suggest to use old good default params pattern that will also properly resolve provided/not provided config options by user. 
> it's not that important for this small mixin but its good idea to ship readable/consistent code


```ts
export const createFocusIndicatorStyleRule = (
  rule: MakeStylesStyleRule<Theme> = defaultStyleRule,
  options?: CreateFocusIndicatorStyleRuleOptions
) => {

  const {selector} = {...defaultOptions,...options}

  return (theme) => ({
    [`${selector}`]: {outline: 'none'},
    [`${KEYBOARD_NAV_SELECTOR} : ${selector}`]:  typeof rule === 'function' ? rule(theme) : rule,
  })
}

```

Reply:
This is actually not right. the user could pass `{selector: undefined}` and that would break the code

Reply:
I don't see why would anyone use this API like that (unless that data type comes from some API service, out of consumers control )

```
createFocusIndicatorStyleRule(...., {selector: undefined});
```

TS 4.4 has this finally covered from type level perspective https://devblogs.microsoft.com/typescript/announcing-typescript-4-4-beta/#exact-optional-property-types